cpp/NamingConventions.md

```diff
@@ -0,0 +1,192 @@
+# BlueBrain HPC Team C++ Naming Conventions
+
+This document describes how C++ symbols should be named.
+
+## Common Conventions
+
+### Use descriptive names
+
+```cpp
+// no
+int bufsz;
+
+// yes
+int buffer_size;
+```
+
+Generally use `snake_case` style as it is compliant with spell checkers.
+
+#### Exception
+
+Function parameters and local variables used only within less than 3 lines can break this rule, for instance:
+
+```cpp
+struct Dimension {
+    const int width;
+    const int height;
+
+    Dimension(int w, int h)
+        : width(w)
+	, height(h) {}
+};
+```
+
+### Functions and variables start with a lower case
+
+```cpp
+void my_function() {
+	int my_var;
+};
+```
+
+### Types names use camel case
+
+```cpp
+class MyClass
+using MyClassVector = std::vector<MyClass>
+```
+
+### Template parameter names use camel case: `InputIterator`
+
+```cpp
+template <class InputIterator, class Distance>
+void advance(InputIterator& it, Distance n);
+```
+
+### Constants are all upper case with underscores
+
+```cpp
+const double AVOGADRO = 6.022140857e23;
+const int LONG_MAX = 9223372036854775807L;
+```
+
+## Distinguish symbols
```

Comment:
This practice i really dislike. I find it makes the code look very ugly. I don't really have an alternative, since I would simply not do it